src/osltoy/osltoyrenderer.cpp

```diff
@@ -78,8 +78,8 @@ OSLToyRenderer::OSLToyRenderer ()
     m_attr_getters[ustring("camera:shutter_close")] = &OSLToyRenderer::get_camera_shutter_close;
 
     // Set up default shaderglobals
-    ShaderGlobals &sg (m_shaderglobals_template);
-    memset (&sg, 0, sizeof(ShaderGlobals));
+    ShaderGlobals &sg = m_shaderglobals_template;
+    sg = {};
```

Comment:
I worry about the efficiency. At best, this will be the same as the memcpy, but it's possible that it will do something less efficient as it fills in fields one by one. I would instead suggest the following simpler fix to the warning:

```suggestion
    memset ((void*)&sg, 0, sizeof(ShaderGlobals));
```

Reply:
I meant memset, not memcpy, of course. My point, though, is that what you are literally asking for is to construct an SG and then copy it where we want it (possibly field by field). We are all hoping that the compiler figures out how to replace that with the equivalent memset to 0, but I feel better about just saying memset explicitly and not taking any chances. Call me old fashioned and superstitious?

Reply:
1. Performance is totally irrelevant here.
2. The compiler is clever enough to avoid double initialization (checked the generated machine code with objdump).

Reply:
Oh yes, I see this is in the constructor. So you're right, there's no perf issue in this spot.

But are you quite sure this is correct? ShaderGlobals doesn't have a constructor defined, so the default constructor will just run the default constructor of the individual fields. But as an example, there are many Vec3 fields, and they are Imath::Vec3, whose default constructor leaves them uninitialized. 

For that matter, `m_shaderglobals_template` will just have had its default initializer run (which I think does nothing), so the `sg = {}` I think is effectively a no-op anyway, is it not? Because you're taking a just-been-default-constructed SG and then overwriting it with a default-constructed SG.

So I'm confused about how all the fields will come to be zeroed out in your proposal. Can you explain?